compiler_opt/rl/problem_configuration.py

```diff
@@ -100,6 +105,143 @@ def get_nonnormalized_features(self) -> Iterable[str]:
   def get_runner_type(self) -> 'type[compilation_runner.CompilationRunner]':
     raise NotImplementedError
 
+  @abc.abstractmethod
+  def get_module_specs(
+      self,
+      data_path: str,
+      additional_flags: Tuple[str, ...] = (),
+      delete_flags: Tuple[str, ...] = ()
+  ) -> List[ModuleSpec]:
+    """Fetch a list of ModuleSpecs for the corpus at data_path
+
+    Args:
+      data_path: base directory of corpus
+      additional_flags: tuple of clang flags to add.
+      delete_flags: tuple of clang flags to remove.
+    """
+    raise NotImplementedError
+
+  @staticmethod
+  def _get_module_specs(data_path: str, additional_flags, delete_flags,
+                        xopts) -> List[ModuleSpec]:
+    module_paths: List[str] = get_module_paths(data_path)
+
```

Comment:
everything below here should probably in another .py file, probably together with the current adt.py (renamed to module_spec.py)

and also need to add unit test for these functions